GenericTensor/Functions/EchelonForm.cs

```diff
@@ -221,20 +237,104 @@ internal static void InnerGaussianEliminationSimple(GenTensor<T, TWrapper> t, in
             }
         }
 
+        internal static void InnerGaussianEliminationSimplePermutation(GenTensor<T, TWrapper> t, int off,
+            ref int[] permute)
```

Comment:
Also, pass `int[]` normally, without `ref`, since the reference itself isn't mutated